src/sage/graphs/graph_decompositions/modular_decomposition.pyx

```diff
@@ -1,27 +1,161 @@
+#                     Corneil-Habib-Paul-Tedder algorithm                      #
+################################################################################
+def corneil_habib_paul_tedder_algorithm(G):
+    r"""
+    Compute the modular decomposition by the algorithm of Corneil, Habib, Paul
+    and Tedder.
+
+    INPUT:
+
+    - ``G`` -- the graph for which modular decomposition tree needs to be
+      computed
+
+    OUTPUT: an object of type Node representing the modular decomposition tree
+    of the graph G
+
+    This function compute the modular decomposition of the given graph by the
+    algorithm of Corneil, Habib, Paul and Tedder [TCHP2008]_. It is a recursive,
+    linear-time algorithm that first computes the slice decomposition of the
+    graph (via the extended lexBFS algorithm) and then computes the modular
+    decomposition by calling itself recursively on the slices of the previously
+    computed slice decomposition.
+
+    .. SEEALSO::
+
+        * :mod:`~sage.graphs.graph_decompositions.slice_decomposition` --
+          compute a slice decomposition of the simple undirect graph
+
+    This function should not be used directly, it should be called via the
+    ``modular_decomposition`` method of ``Graph`` with the parameter
+    ``algorithm='corneil_habib_paul_tedder'``.
+
+    This functions assumes that ``graph`` is a object of the class ``Graph`` and
+    is a simple graph.
+
+    TESTS::
+
+        sage: from sage.graphs.graph_decompositions.modular_decomposition import *
+        sage: recreate_decomposition(15, corneil_habib_paul_tedder_algorithm,
+        ....:                                                         3, 4, 0.2)
+        sage: recreate_decomposition(10, corneil_habib_paul_tedder_algorithm,
+        ....:                                                         4, 5, 0.2)
+        sage: recreate_decomposition(3, corneil_habib_paul_tedder_algorithm,
+        ....:                                                         6, 5, 0.2)
+    """
+    cdef CGraphBackend Gbackend = <CGraphBackend> G._backend
+    cdef CGraph cg = Gbackend.cg()
+
+    cdef vector[int] sigma
+    cdef vector[vector[int]] lex_label
+    cdef vector[size_t] xslice_len
+
+    # Compute the slice decomposition using the extended lexBFS algorithm
+    extended_lex_BFS(cg, sigma, NULL, -1, NULL, &xslice_len, &lex_label)
+
+    cdef SDData SD
+    SD.set_from_data(0, sigma.data(), xslice_len.data(), lex_label.data())
+
+    MD = corneil_habib_paul_tedder_inner(SD)
+
+    r = md_tree_node_to_md_tree(MD, Gbackend)
+    dealloc_md_tree_nodes_recursively(MD)
+    return r
+
+
+cdef object _md_tree_node_to_md_tree_inner_rec(const md_tree_node *n,
+                                               CGraphBackend Gb):
+    """
+    Utility function for :func:`md_tree_node_to_md_tree`.
+    """
+    cdef md_tree_node *c
+    if deref(n).is_leaf():
+        return Node.create_leaf(Gb.vertex_label(deref(n).vertex))
+    else:
```

Comment:
you can remove the `else` and unindent following lines.

Reply:
done in [451b325](https://github.com/sagemath/sage/pull/39038/commits/451b3256ac2cbbb00d2ff24ecc80339fe73d5e40)